assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -124,6 +127,156 @@ export default {
           }
         });
       });
+      api.addSidebarSection(
+        (BaseCustomSidebarSection, BaseCustomSidebarSectionLink) => {
+          const SidebarChatSectionLink = class extends BaseCustomSidebarSectionLink {
+            constructor({ channel }) {
+              super(...arguments);
+              this.channel = channel;
+            }
+
+            get name() {
+              return this.channel.chatable_id;
+            }
+
+            get route() {
+              return "chat.channel";
+            }
+
+            get model() {
+              return {
+                chatChannel: this.channel,
+                channelId: this.channel.id,
+                channelTitle: this.channel.chatable.slug,
+              };
+            }
+
+            get title() {
+              return this.channel.title;
+            }
+
+            get text() {
+              return this.channel.title;
+            }
+
+            get prefixIcon() {
+              return "hashtag";
+            }
+
+            get prefixIconColor() {
+              return this.channel.chatable.color;
+            }
+
+            get prefixIconBadge() {
+              return this.channel.chatable.read_restricted && "lock";
+            }
+
+            get suffixIcon() {
+              return this.channel.unread_count > 0 && "circle";
+            }
+
+            get suffixCssClass() {
+              return this.channel.unread_mentions > 0 ? "urgent" : "unread";
+            }
+          };
+
+          const SidebarChatSection = class extends BaseCustomSidebarSection {
+            @tracked sectionLinks = A([]);
+
+            constructor({ sidebar }) {
+              super(...arguments);
+
+              this.sidebar = sidebar;
+              this.chatService = container.lookup("service:chat");
+              this.sidebar.appEvents.on(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.sidebar.appEvents.on(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+              this.currentUser = getOwner(this).lookup("current-user:main");
+            }
+
+            willDestroy() {
+              this.sidebar.appEvents.off(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.sidebar.appEvents.off(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+            }
+
+            @bind
+            _refreshChannels() {
+              const newSectionLinks = [];
+              this.chatService.getChannels().then((channels) => {
+                channels.publicChannels.forEach((channel) => {
+                  const trackingState =
```

Comment:
Im not sure why we need to have this logic here? This should probably be something we only do in core-chat when we refresh channels. Also channel is an ember object (should be at least) we shouldn't set properties like a POJO.

Reply:
I fixed it, problem was that publicChannels `unread_mentions` and `unread_count` properties were getting out of sync from trackinState. Public channels had initial value and only trackingState was updated. I changed service to update those numbers as well. After that, I could completely drop this logic.